ci/docker/Dockerfile.alpine-release

```diff
@@ -0,0 +1,51 @@
+FROM alpine:3.15 as build
+
+# creates /opensmtpd dir and makes all following commands to run in it
+# https://docs.docker.com/engine/reference/builder/#workdir
+WORKDIR /opensmtpd
+
+# install necessary packages
+RUN apk add --no-cache \
+    autoconf \
+    automake \
+    bison \
+    ca-certificates \
+    fts-dev \
+    gcc \
+    fts \
+    libevent-dev \
+    libtool \
+    libtool \
+    linux-pam-dev \
+    make \
+    musl-dev \
+    openssl \
+    openssl-dev \
```

Comment:
not sure how easy is to run with libressl on alpine but I'd say that if we start providing an "official" dockerfile it should be linked to libressl, by default at least.  The CI is fine using OpenSSL as it's for testing.

This also brings the question of how to automatically push a new image when OpenSSL / LibreSSL gets updated.  The other dependencies are not so important, but updates on OpenSSL and LibreSSL (and maybe libevent) should be applied as soon as possible.

Reply:
Ah, `libbsd` should be added to mix as well, and hopefully the image be re-pushed again when also libbsd gets updated.

Reply:
I can investigate `libressl`, admittedly I am a Linux (or should I say "GNU/Linux") user and have never considered alternatives to openssl.

Reply:
As far as auto-triggering goes, we would need some signal from somewhere that a new `libressl` is available in alpine. GH actions does support triggering actions through an API/from other actions.

Reply:
Actually, there's a small issue with libressl at the moment (OpenSMTPD does a few tricks not so many applications do), so let's keep openssl.  We can revisit it later.  Please discard this point.